src/Form/ChoiceProvider/CMSPageChoiceProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\LinkList\Form\ChoiceProvider;
+
+use Doctrine\DBAL\Connection;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+
+/**
+ * Class CMSPageChoiceProvider
+ * @package PrestaShop\Module\LinkList\Form\ChoiceProvider
+ */
+final class CMSPageChoiceProvider extends AbstractDatabaseChoiceProvider
+{
+    /**
+     * @var array
+     */
+    private $categories;
+
+    /**
+     * CMSPageChoiceProvider constructor.
+     * @param Connection $connection
+     * @param string     $dbPrefix
+     * @param array      $categories
+     * @param int        $idLang
+     * @param int        $idShop
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        array $categories,
+        $idLang,
+        $idShop
+    ) {
+        parent::__construct($connection, $dbPrefix, $idLang, $idShop);
```

Comment:
You should try on PHP 7.2, you may get en error saying parent method parameters don't match. :/

Reply:
Only if it's defined in the interface. If it's only the parent you can override signature.